presentation/src/main/java/com/madrid/presentation/viewModel/homeViewModel/HomeViewModel.kt

```diff
@@ -199,6 +213,27 @@ class HomeViewModel @Inject constructor(
             },
             onError = { onError() }
         )
+        Log.d("tag trailer", "lol yabaaaaaaaaaaaa: ")
+    }
+
+    override fun onClickPlayButton(mediaIndex: Int, context: Context, isMovie: Boolean) {
```

Comment:
passing context to viewmodel method is Violation of Separation of Concerns
and it causes memory leak that why ⬇️
`ViewModels are designed to outlive the Activity or Fragment they are associated with, surviving configuration changes like screen rotations. If a ViewModel holds a direct reference to an Activity or Fragment Context, it can prevent that Context from being garbage collected when the Activity or Fragment is destroyed, leading to a memory leak.`